src/mongo_csfle-v1.h

```diff
@@ -0,0 +1,319 @@
+ * call mongo_csfle_v1_analyze_query.
+ */
+typedef struct mongo_csfle_v1_query_analyzer mongo_csfle_v1_query_analyzer;
+
+/**
+ * Creates a mongo_csfle_v1_query_analyzer object, which stores a parsed collation.
+ *
+ * The 'lib' parameter must be a valid mongo_csfle_v1_lib instance and must not be NULL.
+ *
+ * On failure, it returns NULL and populates the 'status' object if it is not NULL.
+ */
+MONGO_CSFLE_API mongo_csfle_v1_query_analyzer* MONGO_API_CALL
+mongo_csfle_v1_query_analyzer_create(mongo_csfle_v1_lib* lib, mongo_csfle_v1_status* status);
+
+/**
+ * Destroys a valid mongo_csfle_v1_query_analyzer object.
+ *
+ * This function is not thread safe, and it must not execute concurrently with any other function
+ * that accesses the collation object being destroyed including those that access a matcher,
+ * projection or update object which references the collation object.
+ *
+ * This function does not report failures.
+ */
+MONGO_CSFLE_API void MONGO_API_CALL
+mongo_csfle_v1_query_analyzer_destroy(mongo_csfle_v1_query_analyzer* analyzer);
+
+
+/**
+ * Analyzes the 'documentBSON' input document which includes a JSON schema for namespace 'ns_str'
+ * and returns a new BSON document that replaces fields that need encryption with BinData
+ * (subtype 6, first byte 0) placeholders.
+ *
+ * The input document must be a valid OP_MSG document supports aggregate, count, delete, distinct,
+ *  explain, find, findAndModify, insert and update.
+ * In addition, the input document must include two additional top-level fields:
+ * - jsonSchema : document - contains a valid JSON schema
+ * - isRemoteSchema : bool - true if schema comes from MongoD as opposed to client-side
+ *
+ * The 'analyzer' and 'documentBSON' parameters must point to initialized objects of their
+ * respective types.
+ *
+ * When the analysis is successful, this function returns non-null value which points to a valid
+ * BSON document and updates bson_len with the length of the BSON document
+ */
+MONGO_CSFLE_API uint8_t* MONGO_API_CALL
+mongo_csfle_v1_analyze_query(mongo_csfle_v1_query_analyzer* analyzer,
+                             const uint8_t* documentBSON,
+                             const char* ns_str,
+                             uint32_t ns_len,
+                             uint32_t* bson_len,
+                             mongo_csfle_v1_status* status);
+
+/**
+ * Free the memory of a BSON buffer returned by mongo_csfle_v1_analyze_query(). This function can be
+ * safely called on a NULL pointer.
+ *
+ * This function can be called at any time to deallocate a BSON buffer and will not invalidate any
+ * library object.
+ */
+MONGO_CSFLE_API void MONGO_API_CALL mongo_csfle_v1_bson_free(uint8_t* bson);
+
+#ifdef __cplusplus
+}  // extern "C"
+#endif
+
+#undef CSFLE_SUPPORT_API
+#pragma pop_macro("CSFLE_SUPPORT_API")
+
+#undef MONGO_API_EXPORT
+#pragma push_macro("MONGO_API_EXPORT")
+
+#undef MONGO_API_IMPORT
+#pragma push_macro("MONGO_API_IMPORT")
```

Comment:
Probably. This file was copied from the csfle build artifact, and isn't strictly necessary, since we only use symbols loaded at runtime, and only need the enum value and forward declarations of the types therein.